src/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -37,12 +33,12 @@ public UriBuilder(string uri)
 
             if (tryUri.IsAbsoluteUri)
             {
-                Init(tryUri);
+                SetFieldsFromUri(tryUri);
             }
             else
             {
                 uri = Uri.UriSchemeHttp + Uri.SchemeDelimiter + uri;
-                Init(new Uri(uri));
```

Comment:
Oh good point.
On the other hand, this function does do the same thing as SetFieldsFromUri, so any bug fixes ported from the full .NET framework are more likely to change the SetFieldsFromUri methods, which the Init method called anyway